raiden/tests/utils/transfer.py

```diff
@@ -40,51 +43,42 @@ def get_channelstate(app0, app1, token_network_identifier) -> NettingChannelStat
     return channel_state
 
 
-def transfer(initiator_app, target_app, token, amount, identifier):
+def transfer(
+        initiator_app: App,
+        target_app: App,
+        token_address: TokenAddress,
+        amount: PaymentAmount,
+        identifier: PaymentID,
+        fee: FeeAmount = 0,
+        timeout: float = 10,
+) -> None:
     """ Nice to read shortcut to make a transfer.
 
-    The transfer is a LockedTransfer and we make sure,
-    all apps are synched. The secret
-    will also be revealed.
+    The transfer is a LockedTransfer and we make sure, all apps are synched.
+    The secret will also be revealed.
     """
+    assert identifier is not None, 'The identifier must be provided'
+    assert isinstance(target_app.raiden.message_handler, WaitForMessage)
+
+    wait_for_unlock = target_app.raiden.message_handler.wait_for_message(
+        Unlock,
+        {'payment_identifier': identifier},
+    )
+
     payment_network_identifier = initiator_app.raiden.default_registry.address
     token_network_identifier = views.get_token_network_identifier_by_token_address(
-        views.state_from_app(initiator_app),
-        payment_network_identifier,
-        token,
+        chain_state=views.state_from_app(initiator_app),
+        payment_network_id=payment_network_identifier,
+        token_address=token_address,
     )
-    payment_status = initiator_app.raiden.mediated_transfer_async(
-        token_network_identifier,
-        amount,
-        target_app.raiden.address,
-        identifier,
-    )
-    assert payment_status.payment_done.wait()
-
-
-def mediated_transfer(
-        initiator_app,
-        target_app,
-        token_network_identifier,
-        amount,
-        fee=0,
-        identifier=None,
-        timeout=10,
-):
-    """ Nice to read shortcut to make a LockedTransfer.
-
-    The secret will be revealed and the apps will be synchronized."""
-    # pylint: disable=too-many-arguments
-
-    payment_status = initiator_app.raiden.mediated_transfer_async(
+    initiator_app.raiden.mediated_transfer_async(
         token_network_identifier=token_network_identifier,
         amount=amount,
         target=target_app.raiden.address,
         identifier=identifier,
         fee=fee,
     )
-    assert payment_status.payment_done.wait(timeout), f'timeout for transfer id={identifier}'
-    gevent.sleep(0.3)  # let the other nodes synch
+    wait_for_unlock.get(timeout=timeout)
```

Comment:
Hummm, this is not synchronizing on the initiator end ... ideally this would have the path and wait for all the unlocks, but for the moment that is a bit too cumbersome. One trandeoff would be to synchronize both on the initiator and the target, and leave the mediators alonge.